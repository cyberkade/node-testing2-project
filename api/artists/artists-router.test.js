const server = require("../server");
const request = require("supertest");
const db = require("../../data/dbConfig");

beforeAll(async () => {
  await db.migrate.rollback();
  await db.migrate.latest();
});

beforeEach(async () => {
  await db.seed.run();
});

afterAll(async () => {
  await db.destroy();
});

describe("[GET] /api/artists", () => {
  let res;
  beforeEach(async () => {
    res = await request(server).get("/api/artists");
  });
  it("responds with 200 OK", async () => {
    expect(res.status).toBe(200);
  });
  it("responds with all (2) the artists", async () => {
    expect(res.body).toHaveLength(2);
  });
  it("responds with the correct data structure", async () => {
    expect(res.body).toMatchObject([
      { artist_id: 1, artist_name: "Mac Miller" },
      { artist_id: 2, artist_name: "Gorillaz" },
    ]);
  });
});

describe("[POST] /api/artist", () => {
  let res;
  beforeEach(async () => {
    res = await request(server)
      .post("/api/artists")
      .send({ artist_name: "Kid Cudi" });
  });
  it("responds with 201 CREATED", async () => {
    expect(res.status).toBe(201);
  });
  it("causes an artist to be added to the db", async () => {
    const artists = await db("artists");
    expect(artists).toHaveLength(3);
  });
  it("responds with the newly created artist", () => {
    expect(res.body).toMatchSnapshot();
  });
});
