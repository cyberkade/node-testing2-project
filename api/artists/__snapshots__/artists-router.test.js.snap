// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[POST] /api/artist responds with the newly created artist 1`] = `
Object {
  "artist_id": 5,
  "artist_name": "Kid Cudi",
}
`;
